### PR TITLE
Calculate  table inserter width

### PR DESCRIPTION
**Pull request by juliaroldi**
Calculate table insert according to the number, width and height of table cells inside a table row. 

resizerWidth

mergedFirstRow

mergedCellTableInsertergif

### CODE REVIEW

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts [hunk 1]
```diff
@@ -64,6 +65,18 @@ export default function createTableInserter(
     }
 }
 
+const calculateWidth = (tr: HTMLElement) => {
+    let width = 0;
+    tr.childNodes.forEach(td => {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:70]: you can use tr.cells, then no need to use safeInstanceOf. It must be TD or TH.

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts [hunk 2]
```diff
@@ -64,6 +65,18 @@ export default function createTableInserter(
...
+        if (safeInstanceOf(td, 'HTMLTableCellElement')) {
+            const tdRect = normalizeRect(td.getBoundingClientRect());
+            const tdWidth = tdRect.right - tdRect.left;
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:73]: consider just use td.getBoundClientRect()?.width || 0

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts [hunk 3]
```diff
@@ -64,6 +65,18 @@ export default function createTableInserter(
...
+            width = width + tdWidth;
+        }
+    });
+    return width;
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:77]: Do we also need to fix vertical inserter if the first row is merged?

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:77]: And in an extreme case, if all rows are merged, what will happen? For example
```html
<table>
 <tr>
  <td rowspan=2>test</td>
  <td rowspan=2>test</td>
</t>
</table>
```

Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:76]: This is the behavior 
AllmergedRows

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:76]: OK.

About the first question, I mean like this:
image

It seems I can't insert column between the first and second column.

Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableInserter.ts:76]: I see, the vertical inserter was only triggered by the first row cells. I added a check  to verify if the hovered cells has merged cells above, if true, also trigger vertical inserter.

**Comment by JiuqingSong** [comment]
Still see some problems with the fix:
table inserter

Here's the reproing snapshot:

```html
<div><table cellspacing="0" cellpadding="1" data-editing-info="{&quot;topBorderColor&quot;:&quot;#ABABAB&quot;,&quot;bottomBorderColor&quot;:&quot;#ABABAB&quot;,&quot;verticalBorderColor&quot;:&quot;#ABABAB&quot;,&quot;hasHeaderRow&quot;:false,&quot;hasFirstColumn&quot;:false,&quot;hasBandedRows&quot;:false,&quot;hasBandedColumns&quot;:false,&quot;bgColorEven&quot;:null,&quot;bgColorOdd&quot;:&quot;#ABABAB20&quot;,&quot;headerRowColor&quot;:&quot;#ABABAB&quot;,&quot;tableBorderFormat&quot;:0,&quot;keepCellShade&quot;:false}" style="border-collapse: collapse; box-sizing: border-box;" id="tableSelected0"><tbody><tr><td style="width: 241px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; height: 57.3438px; word-break: break-word; white-space: normal;" scope="" colspan="3"><div><br></div><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 57.3438px;" scope=""><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 57.3438px;" scope=""><br></td></tr><tr><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; height: 57.3438px; word-break: break-word; white-space: normal;" scope="" rowspan="2"><br><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; height: 57.3438px; word-break: break-word; white-space: normal;" scope="" rowspan="2"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 28.6719px;"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 28.6719px;"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 28.6719px;"><br></td></tr><tr><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 28.6719px;"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 28.6719px;"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 28.6719px;"><br></td></tr><tr><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; height: 57.3438px; word-break: break-word; white-space: normal;" scope=""><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; height: 57.3438px; word-break: break-word; white-space: normal;" scope=""><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 57.3438px;"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 57.3438px;"><br></td><td style="width: 120.5px; border-width: 1px; border-style: solid; border-color: rgb(171, 171, 171); background-color: transparent; box-sizing: border-box; word-break: break-word; white-space: normal; height: 57.3438px;"><br></td></tr></tbody></table></div><div><br></div><div><br></div><div><br></div><!--{"type":0,"isDarkMode":false,"start":[0,0,0,0,0,0,0],"end":[0,0,0,0,0,0,0]}-->
```

**Comment by BryanValverdeU** [comment]
Checking Word behavior, if the first cell of the row or column is merged, the Row/Column Inserter is not displayed. Not sure we would like to follow that behavior

**Pull request closed without merging**